osquery/utils/azure/azure_util.cpp

```diff
@@ -0,0 +1,107 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <boost/property_tree/json_parser.hpp>
+
+#include <mutex>
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+
+#include <osquery/filesystem/filesystem.h>
+#include <osquery/remote/http_client.h>
+#include <osquery/utils/azure/azure_util.h>
+
+namespace pt = boost::property_tree;
+namespace http = osquery::http;
+namespace fs = boost::filesystem;
+
+namespace osquery {
+
+static bool isAzureInstance() {
```

Comment:
This feels fragile. But I don't have any good advice. I wonder if we should hit the metadata API, and try to parse whether it's ec2 or azure.

Reply:
Yeah, I'm not confident about a best approach here -- it felt overkill (and potentially fragile, given that future Azure tables may not need or have network access) to hit the network and determine the instance type based on metadata response. Then again, testing for a random path on disk isn't much better 😞